src/display/api.js

```diff
@@ -313,52 +314,60 @@ function getDocument(src) {
       if (task.destroyed) {
         throw new Error("Loading aborted");
       }
-      return _fetchDocument(worker, params, rangeTransport, docId).then(
-        function (workerId) {
-          if (task.destroyed) {
-            throw new Error("Loading aborted");
-          }
 
-          let networkStream;
-          if (rangeTransport) {
-            networkStream = new PDFDataTransportStream(
-              {
-                length: params.length,
-                initialData: params.initialData,
-                progressiveDone: params.progressiveDone,
-                disableRange: params.disableRange,
-                disableStream: params.disableStream,
-              },
-              rangeTransport
-            );
-          } else if (!params.data) {
-            networkStream = createPDFNetworkStream({
-              url: params.url,
+      const workerIdPromise = _fetchDocument(
+        worker,
+        params,
+        rangeTransport,
+        docId
+      );
+      const networkStreamPromise = new Promise(function (resolve) {
+        let networkStream;
+        if (rangeTransport) {
+          networkStream = new PDFDataTransportStream(
+            {
               length: params.length,
-              httpHeaders: params.httpHeaders,
-              withCredentials: params.withCredentials,
-              rangeChunkSize: params.rangeChunkSize,
+              initialData: params.initialData,
+              progressiveDone: params.progressiveDone,
               disableRange: params.disableRange,
               disableStream: params.disableStream,
-            });
-          }
-
-          const messageHandler = new MessageHandler(
-            docId,
-            workerId,
-            worker.port
+            },
+            rangeTransport
           );
-          messageHandler.postMessageTransfers = worker.postMessageTransfers;
-          const transport = new WorkerTransport(
-            messageHandler,
-            task,
-            networkStream,
-            params
-          );
-          task._transport = transport;
-          messageHandler.send("Ready", null);
+        } else if (!params.data) {
+          networkStream = createPDFNetworkStream({
+            url: params.url,
+            length: params.length,
+            httpHeaders: params.httpHeaders,
+            withCredentials: params.withCredentials,
+            rangeChunkSize: params.rangeChunkSize,
+            disableRange: params.disableRange,
+            disableStream: params.disableStream,
+          });
         }
-      );
+        resolve(networkStream);
+      });
+
+      return Promise.all([workerIdPromise, networkStreamPromise]).then(data => {
+        if (task.destroyed) {
+          throw new Error("Loading aborted");
+        }
+
+        const messageHandler = new MessageHandler(
+          docId,
+          /* workerId = */ data[0],
```

Comment:
Can we use array destructuring for `data` above so we don't need the inline comments here and below to make it self-documenting? Something like `.then(([workerId, networkStream]) => {` should work I think.

Reply:
Fair point, consider it done :-)